projects/clas12/targets/geometry.py

```diff
@@ -504,13 +461,1169 @@ def build_geometry_pol_targ(configuration):
+		r_in = 0.0 # radius in mm
+		r_out = target_radius # radius in mm
+		half_length = target_window_thickness/2  # half length along beam axis
+		gvolume = GVolume("NH3DSWindow")
+		gvolume.mother = "HeliumBath"
+		gvolume.description = "NH3 Target cup Downstream Window"
+		gvolume.setPosition(0,0,z_center)
+		gvolume.color = "aaaaaa"
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_Al"
+		return gvolume
+
+	def build_beam_pipe():
+		# Beam pipe
+		r_in  = 0.0
+		r_out = target_radius+1
+		z1 = -bath_dz
+		z2 = -bath_z0-target_length/2-target_window_thickness
+		half_length = (z2-z1)/2
+		z_center    = (z2+z1)/2
+		gvolume = GVolume("BeamEntranceVacuum")
+		gvolume.mother = "HeliumBath"
+		gvolume.description = "Beam Entrance Vacuum"
+		gvolume.color = "ffffff"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_Galactic"
+		return gvolume
+
+	def build_beam_entrance_pipe():
+		r_in  = target_radius+1
+		r_out = target_radius+2
+		z1 = -bath_dz
+		z2 = -bath_z0-target_length/2-target_window_thickness
+		half_length = (z2-z1)/2
+		z_center     = (z2+z1)/2
+		gvolume = GVolume("BeamEntrancePipe")
+		gvolume.mother = "HeliumBath"
+		gvolume.description = "Beam Entrance Pipe"
+		gvolume.color = "595959"
+		gvolume.setPosition(0,0,z_center)
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_Al"
+		return gvolume
+
+	def build_beam_entrance_window():
+		#  Beam Entrance Window
+		z1 = -bath_dz
+		z2 = -bath_z0-target_length/2-target_window_thickness
+		z_half_length = (z2-z1)/2
+		z_center = z_half_length-beam_window_thickness/2
+		r_in = 0.0      # radius in mm
+		r_out = target_radius+1    # radius in mm
+		half_length = beam_window_thickness/2  # half length along beam axis
+		gvolume = GVolume("BeamEntrance")
+		gvolume.mother = "BeamEntranceVacuum"
+		gvolume.description = "Beam entrance window"
+		gvolume.setPosition(0,0,z_center)
+		gvolume.color = "aaaaaa"
+		_make_full_tube(gvolume, r_in, r_out, half_length)
+		gvolume.material = "G4_Al"
+		return gvolume
+
+	def build_shim_up_upstream_coil():
+		return _create_shimcoil(bath_half_length - shim_coils_length / 2, 1)
+
+	def build_shim_upstream_coil():
+		return _create_shimcoil(-4.5, 2)
+
+	def build_shim_downstream_coil():
+		return _create_shimcoil(-22.5, 3)
+
+	def build_shim_coil_carrier():
+		yield from _create_spheretube_volumes(
+			shim_coils_mandrel_r_in,
+			shim_coils_mandrel_r_out,
+			shim_coils_window,
+			"ShimCoil",
+			"G4_Al",
+			"aaaaaa"
+		)
+
+	def build_pumping_volume():
+		yield from _create_spheretube_volumes(
+			pumping_volume_r_in,
+			pumping_volume_r_out,
+			pumping_volume_window,
+			"PumpingVolume",
+			"G4_Al",
+			"000080"
+		)
+
+	def build_heat_shield_volume():
+		yield from _create_spheretube_volumes(
+			heat_shield_r_in,
+			heat_shield_r_out,
+			heat_shield_window,
+			# @mariakzurek: should this be "HeatShield"?
```

Comment:
@maureeungaro Please check this comment

Reply:
Yes I'd leave the original name. Can always change later.